File: app-playfab/src/main/java/com/xsolla/android/storesdkexample/adapter/InventoryAdapter.java
package com.xsolla.android.storesdkexample.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.xsolla.android.storesdkexample.R;
import com.xsolla.android.storesdkexample.data.store.Store;
import com.xsolla.android.storesdkexample.listener.ConsumeListener;
import com.xsolla.android.storesdkexample.util.ViewUtils;

import org.jetbrains.annotations.NotNull;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class InventoryAdapter extends RecyclerView.Adapter<InventoryAdapter.ViewHolder> {

    private List<Store.InventoryItem> items;
    private ConsumeListener consumeListener;

    public InventoryAdapter(List<Store.InventoryItem> items, ConsumeListener consumeListener) {
        this.items = items;
        this.consumeListener = consumeListener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_inventory, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        ImageView itemIcon;
        TextView itemName;
        TextView itemExpiration;
        Button consumeButton;
        TextView quantityLabel;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            itemIcon = itemView.findViewById(R.id.item_icon);
            itemName = itemView.findViewById(R.id.item_name);
            itemExpiration = itemView.findViewById(R.id.item_expiration);
            consumeButton = itemView.findViewById(R.id.consume_button);
            quantityLabel = itemView.findViewById(R.id.quantity_label);
        }

        public void bind(Store.InventoryItem item) {

            Glide.with(itemView).load(item.getImageUrl()).into(itemIcon);
            itemName.setText(item.getName());

            itemExpiration.setVisibility(View.GONE);

            if (item.getRemainingUses() == 0) {
                quantityLabel.setVisibility(View.GONE);
                consumeButton.setVisibility(View.GONE);
                return;
            } else {
                quantityLabel.setVisibility(View.VISIBLE);
                consumeButton.setVisibility(View.VISIBLE);
            }

            quantityLabel.setText(String.valueOf(item.getQuantity()));
            consumeButton.setOnClickListener(v -> {
                ViewUtils.disable(v);

                Store.consumeItem(item.getInstanceId(), 1, new Store.ConsumeCallback() {
                    @Override
                    public void onSuccess() {
                        consumeListener.onSuccess();
                        Store.getInventory(new Store.InventoryCallback() {
                            @Override
                            public void onSuccess(@NotNull List<Store.InventoryItem> inventoryItems) {
                                items = inventoryItems;
                                notifyDataSetChanged();
                                ViewUtils.enable(v);
                            }

                            @Override
                            public void onFailure(@NotNull String errorMessage) {
                                consumeListener.onFailure(errorMessage);
                                ViewUtils.enable(v);
                            }
                        });
                    }

                    @Override
                    public void onFailure(@NotNull String errorMessage) {
                        consumeListener.onFailure(errorMessage);
                        ViewUtils.enable(v);
                    }
                });
            });
        }
    }

}

File: app-inventory/src/main/java/com/xsolla/android/inventorysdkexample/adapter/InventoryAdapter.kt
package com.xsolla.android.inventorysdkexample.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.view.isVisible
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.xsolla.android.appcore.databinding.ItemInventoryBinding
import com.xsolla.android.inventory.entity.response.InventoryResponse
import com.xsolla.android.inventory.entity.response.SubscriptionsResponse
import com.xsolla.android.inventorysdkexample.R
import com.xsolla.android.inventorysdkexample.ui.fragments.store.ConsumeListener
import java.text.SimpleDateFormat
import java.util.*

class InventoryAdapter(
    var items: List<InventoryResponse.Item>,
    private val consumeListener: ConsumeListener
) : RecyclerView.Adapter<InventoryAdapter.ViewHolder>() {

    private var subscriptions: List<SubscriptionsResponse.Item>? = null

    fun setSubscriptions(subscriptions: List<SubscriptionsResponse.Item>?) {
        this.subscriptions = subscriptions
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return ViewHolder(inflater, parent)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = items[position]
        holder.bind(item)
    }

    override fun getItemCount() = items.size


    inner class ViewHolder(
        inflater: LayoutInflater,
        parent: ViewGroup
    ) : RecyclerView.ViewHolder(inflater.inflate(R.layout.item_inventory, parent, false)) {

        private val binding: ItemInventoryBinding = ItemInventoryBinding.bind(itemView)

        fun bind(item: InventoryResponse.Item) {
            Glide.with(itemView).load(item.imageUrl).into(binding.itemIcon)
            binding.itemName.text = item.name
            binding.itemQuantity.text = item.quantity.toString()
            if (item.virtualItemType == InventoryResponse.Item.VirtualItemType.NON_RENEWING_SUBSCRIPTION) {
                binding.itemQuantity.visibility = View.INVISIBLE
            } else {
                binding.itemQuantity.visibility = View.VISIBLE
            }

            val subscription =
                if (item.virtualItemType == InventoryResponse.Item.VirtualItemType.NON_RENEWING_SUBSCRIPTION) {
                    subscriptions?.find { it.sku == item.sku }
                } else {
                    null
                }

            binding.itemExpiration.text = getExpirationText(subscription)

            binding.consumeButton.visibility =
                if (item.remainingUses == null || item.remainingUses == 0L) View.INVISIBLE else View.VISIBLE
            binding.consumeButton.setOnClickListener { consumeListener.onConsume(item) }

            binding.buyAgainButton.isVisible = false
        }

        private fun getExpirationText(sub: SubscriptionsResponse.Item?): String? {
            if (sub == null) return null
            return if (sub.status == SubscriptionsResponse.Item.Status.ACTIVE) {
                val date = Date(sub.expiredAt?.times(1000)!!)  //date *1000
                val sdf = SimpleDateFormat("MM/dd/yyyy hh:mm:ss a", Locale.US)
                val formattedDate = sdf.format(date)
                "Active until: $formattedDate"
            } else {
                "Expired"
            }
        }

    }
}